package-res/resources/web/pentaho/data/Filter.js

```diff
@@ -22,8 +22,22 @@ define([
 ], function(IsEqual, IsIn, AndFilter, OrFilter, NotFilter) {
   "use strict";
 
-
-  var RootFilter = OrFilter.extend({
+  /**
+   * @name RootFilter
+   * @memberOf pentaho.data.filter
+   * @class
+   * @ignore
+   *
+   * @classdesc The `RootFilter` class implements a type of
+   * `AbstractTreeFilter` {@link pentaho.data.filter.AbstractTreeFilter}. Specifically
```

Comment:
The tag @extends already provides this information.
I think the @classdesc tag should explain the what and the why and not so much the how.

The RootFilter class is not really exposed, and is mostly an artifice to create a filter tree from a spec.